model/src/main/kotlin/utils/PurlUtils.kt

```diff
@@ -92,28 +93,44 @@ internal fun createPurl(
 ): String =
     buildString {
         append("pkg:")
-        append(type)
+        append(type.lowercase())
+        append('/')
 
         if (namespace.isNotEmpty()) {
+            append(namespace.trim('/').split('/').joinToString("/") { it.percentEncode() })
             append('/')
-            append(namespace.percentEncode())
+            append(name.trim('/').percentEncode())
```

Comment:
Could the `else` be dropped and line 102 be executed in any case (outside the `if`) ?

Reply:
oh, it's to be more close to the algo description. Then just ignore my comment.

Reply:
Let's keep it now that I've changed it.